egs2/chime7_task1/asr1/run.sh

```diff
@@ -142,34 +152,34 @@ if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
     local/run_gss.sh --manifests-dir $manifests_root --dset-name $dset_name \
           --dset-part $dset_part \
           --exp-dir $gss_dump_root \
-          --cmd $cmd_gss \
+          --cmd "$cmd_gss" \
           --nj $ngpu \
           --max-segment-length $max_segment_length \
           --max-batch-duration $gss_max_batch_dur \
           --channels $channels \
-          --use-selection $use_selection
+          --use-selection $use_selection \
+          --top-k $top_k
     log "Guided Source Separation processing for ${dset_name}/${dset_part} was successful !"
   done
 fi
 
 
 if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ]; then
-
-  asr_train_set=kaldi/train_all_mdm_ihm_rvb_gss
-  asr_cv_set=kaldi/chime6/dev/gss # use chime only for validation
   # Decoding on dev set because test is blind for now
-  # NOTE that ESPNet will not make copies of the original Kaldi manifests
-  # e.g. for training and cv, so if you set $train_max_segment_length these
-  # will be discarded also from the test set (if the test set is the same as evaluation)
-  # you need to make a copy !
-  ./utils/copy_data_dir.sh data/kaldi/chime6/dev/gss data/kaldi/chime6/dev/gss_inf
-  asr_tt_set="kaldi/chime6/dev/gss_inf kaldi/dipco/dev/gss/ kaldi/mixer6/dev/gss/"
 
   pretrained_affix=
   if [ -n "$use_pretrained" ]; then
-    asr_train_set=kaldi/dipco/dev/gss # dummy one, it is not used
+    asr_train_set="dummy_tr" # dummy one, it is not used
+    asr_cv_set="dummy_cv"
     pretrained_affix+="--skip_data_prep false --skip_train true "
     pretrained_affix+="--download_model ${use_pretrained}"
+  else
+    asr_train_set=kaldi/train_all_mdm_ihm_rvb_gss
+    asr_cv_set=kaldi/chime6/dev/gss # use chime only for validation
+    # we will make a copy to avoid ESPNet modify this as it will discard
+    # utterances longer than $train_max_segment_length.
+    ./utils/copy_data_dir.sh ./data/$asr_cv_set ./data/${asr_cv_set}_discard_long
```

Comment:
@kamo-naoyuki this copy should fix the issue with the discarding no ? 
(I would avoid like to avoid to change the parsing in the scoring afterwards due to change in the path with the data/org).

Reply:
Actually IMO it would be better if the cv file path is changed and not the path for the inference: instead of data/org/tt_dataset maybe data/mod/cv_dataset

Reply:
Originally, it was not right way to use a test set for the validation set of DNN  because DNN uses the lr tuning the validation set. The current result is biased by the `chime6/dev`.  We sometimes split the training set and create a new validation set in the recipe. This is why we didn't expect your usage. 

> Actually IMO it would be better if the cv file path is changed and not the path for the inference: instead of data/org/tt_dataset maybe data/mod/cv_dataset

> this copy should fix the issue with the discarding no ?

Reply:
There is another dev set actually created in the recipe ! it is a concatenation of all devs enhanced by GSS it is called `data/kaldi/dev_all_gss`.

Reply:
> Actually IMO it would be better if the cv file path is changed and not the path for the inference: instead of data/org/tt_dataset maybe data/mod/cv_dataset

I agree, but please let me. We need to change many to do this. 

>     ./utils/copy_data_dir.sh ./data/$asr_cv_set ./data/${asr_cv_set}_discard_long
> this copy should fix the issue with the discarding no ?

Sorry, I missed it. 

This is acceptable, but extra files are generated, so we should avoid it. I don't know why you stick to change `asr_train_set` and `asr_cv_set `. There are many other solutions. e.g. the following easily solve this problem.

```sh
for tt_dset in $asr_tt_set; do
   [ -d  ${asr_exp}/${inference_tag}/org/${tt_dset} ]
   ln -s ${asr_exp}/${inference_tag}/org/${tt_dset} ${asr_exp}/${inference_tag}/${tt_dset*
done  
```

Reply:
That also can do it. Okay let me change

Reply:
I am currently re-testing inference. I will ping you when everything runs smoothly

Reply:
> There is another dev set actually created in the recipe ! it is a concatenation of all devs enhanced by GSS it is called data/kaldi/dev_all_gss.

This is worse than use chime6/dev. DNN trainig can knows all dev sets.

The best is splitting the trainig set.

Reply:
I don't think it will make a huge difference honestly. At the end you always tune on the dev set. 
E.g. you early stop on the development set because it is supposed to be representative of evaluation. 
But yes making an additional split may reduce risks of overfitting on development. 
I followed the old kaldi recipes.

Reply:
Sorry, for pointing many and pushing them. This is jut my philosophy.